### PR TITLE
feat: enable auto-population of missing metadata in logs and printing structured logs to stdout

### LINKED ISSUE #689: structured logging: opt-out automatic population of metadata when writing log entries
**Is your feature request related to a problem? Please describe.**
Support the mode of work when logging library does not auto-populate log entries with additional metadata details.

**Describe the solution you'd like**
All fields except to payload of all log entries passed to the `write()` API call will be auto-populated unless they already have values. A caller should be able to opt-out this behavior by providing a `WriteOption.autoPopulateMetadata()` flag or by using `LoggingOptions.Builder.setAutoPopulateMetadata()` to opt-out on the level of the `Logging` instance.

### LINKED ISSUE #691: structured logging: auto-populate empty fields in LogEntry collection on call to Logging.write() interface
**Is your feature request related to a problem? Please describe.**
Fields in LogEntry structure are used to provide a rich user experience when using Log Explorer and other tools of the Cloud Operations suites. When developer does not explicitly populate the fields the additional functionality like tracing, log correlation and many others aren't available.

**Describe the solution you'd like**
To improve user experience the additional fields in LogEntry structure should be auto-populated. The fields that should be populated are:
- resource
- httpRequest
- severity
- trace
- spanId
- sourceLocation
- ~metadata~

### LINKED ISSUE #799: structured logging: Support to opt-out metadata population in LoggingHandler
Extend opt-out configurations (#689) to the JUL handler class `LoggingHandler`.

### LINKED ISSUE #800: structured logging: configure LoggingHandler to print to STDOUT instead of calling Logging API
Add system property to `LoggingHandler` to allow it to print to STDOUT in the form of one line stringified JSON (a.k.a. structured logging print out) instead of calling `Logging.write()` API.

### PR DESCRIPTION
**Pull request by minherz**
PR aggregates the following changes:

- https://github.com/googleapis/java-logging/pull/821
- https://github.com/googleapis/java-logging/pull/812
- https://github.com/googleapis/java-logging/pull/807
- https://github.com/googleapis/java-logging/pull/803
- https://github.com/googleapis/java-logging/pull/798

Fixes #689, closes #691, closes #799 and resolves #800

### CODE REVIEW

File: google-cloud-logging/src/main/java/com/google/cloud/logging/LogEntry.java [hunk 1]
```diff
@@ -575,6 +583,142 @@ public Builder toBuilder() {
...
+     * @param name a valid Json field name string.
+     * @param value an object to be serialized to Json using {@link Gson}.
+     * @param appendComma a flag to add a trailing comma.
+     * @return a reference to this object.
+     */
+    public StructuredLogFormatter appendField(String name, Object value, boolean appendComma) {
+      checkNotNull(name);
+      if (value != null) {
+        builder.append(gson.toJson(name)).append(":").append(gson.toJson(value));
+        if (!appendComma) {
```
Comment (losalex) [inline google-cloud-logging/src/main/java/com/google/cloud/logging/LogEntry.java:656]: Hm... How comes it contains a bug with `if (!appendComma)`? Is thats a latest merge?

File: google-cloud-logging/src/main/java/com/google/cloud/logging/LogEntry.java [hunk 2]
```diff
@@ -575,6 +583,142 @@ public Builder toBuilder() {
...
+     * @param appendComma a flag to add a trailing comma.
+     * @return a reference to this object.
+     */
+    public StructuredLogFormatter appendDict(Map<String, Object> value, boolean appendComma) {
+      if (value != null) {
+        String json = gson.toJson(value);
+        // append json object without brackets
+        if (json.length() > 1) {
+          builder.append(json.substring(0, json.length() - 1).substring(1));
+          if (!appendComma) {
```
Comment (losalex) [inline google-cloud-logging/src/main/java/com/google/cloud/logging/LogEntry.java:680]: Same here... `if (!appendComma)`

**Review by losalex — COMMENTED** [review]
It seems that you didnt merge latest changes which were discussed in previous iterations, can you please take a look and make sure all latest issues are addressed?

File: google-cloud-logging/src/main/java/com/google/cloud/logging/LogEntry.java (hunk 1 above)
Comment (minherz) [reply google-cloud-logging/src/main/java/com/google/cloud/logging/LogEntry.java:656]: It looks like the whole #814 did not merge into the branch... I will look into it

Comment (minherz) [reply google-cloud-logging/src/main/java/com/google/cloud/logging/LogEntry.java]: I have re-created the PR (new one is #821) which adds all the changes

File: google-cloud-logging/src/main/java/com/google/cloud/logging/LogEntry.java (hunk 2 above)
Comment (minherz) [reply google-cloud-logging/src/main/java/com/google/cloud/logging/LogEntry.java]: idem hic

**Review by losalex — APPROVED** [review]

**Pull request merged**